src/core/geometry/qgsgeometry.cpp

```diff
@@ -1595,6 +1595,12 @@ double QgsGeometry::distance( const QgsGeometry &geom ) const
     return -1.0;
   }
 
+  // avoid calling geos for trivial point-to-point distance calculations
+  if ( QgsWkbTypes::flatType( d->geometry->wkbType() ) == QgsWkbTypes::Point && QgsWkbTypes::flatType( geom.constGet()->wkbType() ) == QgsWkbTypes::Point )
+  {
+    return qgsgeometry_cast< QgsPoint * >( d->geometry.get() )->distance( *qgsgeometry_cast< const QgsPoint * >( geom.constGet() ) );
```

Comment:
`distance()` can be called on a const point as well, so better make it a `constGet()`, no reason to detach.

Reply:
@m-kuhn , I'm being told by my console that `d->geometry` doesn't have a `constGet()` function.

Reply:
hah, you are of course right :)